.github/workflows/rust.yml

```diff
@@ -35,14 +42,48 @@ jobs:
         rustflags:
 
     - name: Install Nextest test runner
-      uses: taiki-e/install-action@nextest
+      run: cargo install cargo-nextest
 
     - name: Create example env file
       run: |
         chmod +x .github/scripts/test_config.sh
         sh .github/scripts/test_config.sh
 
-    - name: Build and Run Test
+    - name: Check Formatting
+      run: cargo fmt --all -- --check
+
+    - name: Build Project
+      run: cargo build --workspace --all-features
+
+    - name: Run Tests
+      run: cargo nextest run --workspace --all-features
+
+  # Continuous Deployment Job
+  cd:
+    name: Continuous Deployment
+    needs: ci 
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Rust
+      uses: actions-rust-lang/setup-rust-toolchain@v1
+      with:
+        toolchain: stable
+
+    - name: Build Release
+      run: cargo build --release --workspace --all-features
+
+    - name: Deploy to Production
+      env:
+        SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
+        HOST: ${{ secrets.PRODUCTION_HOST }}
+        USER: ${{ secrets.PRODUCTION_USER }}
       run: |
-        cargo build --workspace --all-features
-        cargo nextest run --workspace --all-features
+        echo "${{ secrets.SSH_PRIVATE_KEY }}" > ssh_key
+        chmod 600 ssh_key
+        scp -i ssh_key target/release/your_binary ${{ env.USER }}@${{ env.HOST }}:/path/to/your/binary
+        ssh -i ssh_key ${{ env.USER }}@${{ env.HOST }} 'systemctl restart your-service'
+        rm -f ssh_key
```

Comment:
Where are you deploying the server to? Aren't we supposed to have an AWS instance set up for the project? I don't get the logic here...

Reply:
We are to Have